sys/dev/proto/proto_busdma.c

```diff
@@ -109,7 +109,7 @@ proto_busdma_tag_create(struct proto_busdma *busdma, struct proto_tag *parent,
 		tag->datarate = ioc->u.tag.datarate;
 	}
 	LIST_INSERT_HEAD(&busdma->tags, tag, tags);
-	ioc->result = (uintptr_t)(void *)tag;
+	ioc->result = (__cheri_addr vaddr_t)tag;
```

Comment:
These could now be ptraddr_t.

Reply:
We could also do `(ptraddr_t)(uintptr_t)(void *)` rather than using the ugly `__cheri_addr`, though maybe others will think the chain of casts is uglier. I do like the chain of casts though because it's not really CHERI-specific (well other than `ptraddr_t`, though using `size_t`/`vm_offset_t`/... generally gives upstreamable patches), and when I get around to finishing my purecap LLVM that's how I intend to upstream these kinds of things.

Reply:
I also prefer `(ptraddr_t)(uintptr_t)(void *)` over `__cheri_addr`. I'd quite like it if we could remove (almost) all `__cheri_addr` casts, since they are just a workaround for offset mode and/or ctoptr not being suitable as a default ptr->int conversion.

Reply:
Surely just `(ptraddr_t)(uintptr_t)`?  That isn't likely upstreamable though as things like `(vm_offset_t)(uintptr_t)` are basically NOPs that upstream won't want, and they wouldn't match what's in CheriBSD anyway.

Reply:
Yeah the `(void *)` is pointless even upstream, that's true.